be/src/vec/data_types/serde/data_type_nullable_serde.cpp

```diff
@@ -121,10 +120,11 @@ Status DataTypeNullableSerDe::deserialize_one_cell_from_json(IColumn& column, Sl
     auto& null_column = assert_cast<ColumnNullable&>(column);
     // TODO(Amory) make null literal configurable
 
+    // only slice trim quote return true make sure slice is quoted and converted_from_string make
+    // sure slice is from string parse , we can parse this "null" literal as string "null" to
+    // nested column , otherwise we insert null to null column
     if (!(options.converted_from_string && slice.trim_quote())) {
-        //for map<string,string> type : {"abc","NULL"} , the NULL is string , instead of null values
-        if (slice.size == 4 && slice[0] == 'N' && slice[1] == 'U' && slice[2] == 'L' &&
-            slice[3] == 'L') {
+        if (Slice::mem_equal("null", slice.data, 4)) {
```

Comment:
why change NULL to null ?

Reply:
slice.size == 4 should be checked since Slice::mem_equal does not check it.

Reply:
in json we do not support NULL just null